org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleBuild.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+/**
+ * Represents a Gradle build that can be imported into or synchronized with the workspace.
+ *
+ * @author Donat Csikos
+ * @since 3.0
+ */
+public interface GradleBuild {
+
+    /**
+     * Synchronizes this build with the workspace.
+     *
+     * @param monitor the progress monitor
```

Comment:
I find comments like these unhelpful. They're just noise, repeating what the types are saying.  If there is nothing else to say about them, why not leave them out?

Reply:
It's kind of obvious what it does, although I'd still prefer to have an entry for all params for a public API.